bin/get_qc_config.py

```diff
@@ -5,16 +5,58 @@
 import argparse
 import yaml
 
-def convert_to_multiqc_config(checkqc_config):
-    for qc_criteria in checkqc_config:
-        if qc_criteria['name'] == 'ClusterPFHandler':
-            bcl2fastq_total = {'total': {}}
-            if not qc_criteria['warning'] == 'unknown':
-                bcl2fastq_total['total']['warn'] = [{'lt': qc_criteria['warning']}]
-            if not qc_criteria['error'] == 'unknown':
-                bcl2fastq_total['total']['error'] = [{'lt': qc_criteria['error']}]
 
-    return {'table_cond_formatting_rules': bcl2fastq_total}
+class ValueHandlerMapper(object):
+    def __init__(self, handler_name, multiqc_mapping, compare_direction):
+        self.handler_name = handler_name
+        self.multiqc_mapping = multiqc_mapping
+        self.compare_direction = compare_direction
+
+
+class HandlerMapper():
+    def __init__(self):
+        self._mapper_list = [ValueHandlerMapper(handler_name = 'ClusterPFHandler',
+                                                multiqc_mapping = 'total',
+                                                compare_direction = 'lt'),
+                             ValueHandlerMapper(handler_name = 'ErrorRateHandler',
+                                                multiqc_mapping = 'Error',
+                                                compare_direction = 'gt'),
+                             ValueHandlerMapper(handler_name = 'Q30Handler',
+                                                multiqc_mapping = 'percent_Q30',
+                                                compare_direction = 'lt'),
+                             ValueHandlerMapper(handler_name = 'ReadsPerSampleHandler',
+                                                multiqc_mapping = 'mqc-generalstats-bcl2fastq-total',
+                                                compare_direction = 'lt')]
+
+        self.mapping = self._convert_to_mappings(self._mapper_list)
+
+    def _convert_to_mappings(self, mapper_list):
+        mapper_dict = {}
+        for mapper in mapper_list:
+            mapper_dict[mapper.handler_name] = mapper
+        return mapper_dict
+
+def convert_to_multiqc_config(checkqc_config_dict):
+    multiqc_config_format = {}
+    handler_mapper = HandlerMapper()
+    for mapper_name, mapper in handler_mapper.mapping.items():
+        qc_criteria = checkqc_config_dict.get(mapper.handler_name)
+        multiqc_config_value = {mapper.multiqc_mapping: {}}
+        if not qc_criteria['warning'] == 'unknown':
+            multiqc_config_value[mapper.multiqc_mapping]['warn'] = [{mapper.compare_direction: qc_criteria['warning']}]
+        if not qc_criteria['error'] == 'unknown':
+            multiqc_config_value[mapper.multiqc_mapping]['fail'] = [{mapper.compare_direction: qc_criteria['error']}]
+
+        multiqc_config_format[mapper.multiqc_mapping] = multiqc_config_value[mapper.multiqc_mapping]
+
+    return {'table_cond_formatting_rules': multiqc_config_format}
+
+def convert_to_dict(checkqc_config):
```

Comment:
I think it would be clearer if this was called something like `convert_checkqc_config_to_dict`.